### DISCUSSION TITLE
different behavior on different page loading

### DISCUSSION THREAD

**Discussion by bubnenkoff** [alpinejs/alpine#2498] [1. Help]
I am doing simple google oAuth-login system. The problem that on dufferent page (re)loading I am getting different `isLogined` status for logined user. I checking status in Chrome console: `Alpine.store('auth')` sometimes `isLogined` is `true` sometimes `false`. UI also show wrong login status.
I can't figure out what's wrong here is my code:
```
<template x-data x-if="!Alpine.store('auth').isLogined">
  <div class="g-signin2" data-onsuccess="onSignIn"></div> 
  </template>
 <template x-data x-if="Alpine.store('auth').isLogined">
  <button x-data @click="logout">Logout</button>  
 </template>
 ```
 ...
 ```
async function init() { // I tried also without async keyword here, same effect
        window.gapi.load('auth2', async function() {
            console.log("Hello we are here");
    /* Ready. Make a call to gapi.auth2.init or some other API */
    window.gapi.auth2.init({
        client_id: '....apps.googleusercontent.com',
    });

    var authInstance = await window.gapi.auth2.getAuthInstance();
    var signedIn = authInstance.isSignedIn.get()
    if (signedIn) {
         console.log(Alpine.store('auth').isLogined);
         console.log('-------------------');
         Alpine.store('auth').isLogined = true;

        var user = authInstance.currentUser.get();
        var profile = user.getBasicProfile();
        console.log('Email: ' + profile.getEmail());


    }
    else {
        Alpine.store('auth').isLogined = false;
        console.log('you are guest');
    }

  });
}
```
auth.js:
```
document.addEventListener('alpine:init', () => {
   Alpine.store('auth', {
        init() {
            console.log("Hello from AlpineJS"); //work fine at page loading
        },
        isLogined: false,

        data: [

        ]
    })
})
```

UPD:
I have found problem. I need to call `init` After alpinejs is initialized. Not I moved call of `init` to `x-init` like:
`<span x-data x-init="init()"></span>` and removed `?onload=init` from:
`<script src="https://apis.google.com/js/platform.js"></script>`
before was:
`<script src="https://apis.google.com/js/platform.js?onload=init" async defer></script>`

Could anybody say where it's better to move `init`? To get it work after alpine is initialized?

I have only one idea like this. It works. But is it good place?
```
document.addEventListener('alpine:init', () => {
   Alpine.store('auth', {
        init() {
            console.log("Hello from AlpineJS");
            initAuth(); // I have only one idea like this
        },
        isLogined: false,

        data: [

        ]
    })
})
```

**Comment by danddanddand** [comment]
- Be careful using x-data with function named init() as Alpine will call it automatically resulting in double calls. Name it anything else if it's in component scope (your example wasn't clear where that would've been defined)
- Using the 'alpine:initialized' lifecycle (https://alpinejs.dev/essentials/lifecycle#alpine-initialized) is probably the right place. If initAuth() is just global function that needs to be called, it doesn't have to live in the store definition (could be in it's own 'alpine:init' listener or 'alpine:initialized'))

**Reply by bubnenkoff** [reply]
So you are suggest to do something like https://gist.github.com/bubnenkoff/f761c8e4b978e62dbe06a569e408ce97 ?

**Reply by danddanddand** [reply]
Yeah, for CDN build that way you know the Store's are all setup and Alpine has finished doing all it's work. 

If you are doing NPM bundled version, you'd just call it after Alpine.start()
